crates/iota-indexer/src/apis/indexer_api.rs

```
@@ -91,6 +91,65 @@ impl IndexerApi {
             has_next_page,
         })
     }
+
+    async fn get_dynamic_field_object(
+        &self,
+        parent_object_id: ObjectID,
+        name: DynamicFieldName,
+        options: Option<IotaObjectDataOptions>,
+    ) -> RpcResult<IotaObjectResponse> {
+        let name_bcs_value = self.inner.bcs_name_from_dynamic_field_name(&name).await?;
+
+        // Try as Dynamic Field
+        let id = iota_types::dynamic_field::derive_dynamic_field_id(
+            parent_object_id,
+            &name.type_,
+            &name_bcs_value,
+        )
+        .expect("deriving dynamic field id can't fail");
```

If it can't fail then use unwrap and add a comment

---

The [std](https://doc.rust-lang.org/std/error/index.html#converting-errors-into-panics) docs suggest otherwise

> Of the two, expect is generally preferred since its msg field allows you to convey your intent and assumptions which makes tracking down the source of a panic easier. unwrap on the other hand can still be a good fit in situations where you can trivially show that a piece of code will never panic, such as "127.0.0.1".parse::<std::net::IpAddr>().unwrap() or early prototyping.

---

Fair enough

---

Though this is exactly the kind of message that you don't want a user to ever see. 😆 it may be better to say "failed to derive dynamic id" and you could also add a comment saying it shouldn't fail, I guss

---

That actually raises another issue. @samuel-rufi why don't we just handle the error as an internal server error?

I understand that this `expect` is used as a [precondition](https://doc.rust-lang.org/std/error/index.html#common-message-styles) but a public endpoint does not seem as a place for panicking and developer debugging.

---

Good point, I was only moving the code in a dedicated function and didn't want to touch the logic or change anything in the error handling. I agree, we should return an internal error to the user and log it additionally.

---

Please see https://github.com/iotaledger/iota/pull/6415